File: java-reactive-spring/src/main/java/com/ahierro/benchmarks/javareactivespring/Controller.java
package com.ahierro.benchmarks.javareactivespring;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

import java.time.Duration;

import static java.time.temporal.ChronoUnit.MILLIS;

@RestController
public class Controller {

    @GetMapping("/v1/benchmark")
    public Mono<String> getHelloSleep(@RequestParam Long delay){
        return Mono.delay(Duration.of(delay, MILLIS))
                .map(result -> "");
    }

}

File: kotlin-spring/src/main/kotlin/com/benchmark/kotlinspring/Controller.kt
package com.benchmark.kotlinspring

import kotlinx.coroutines.delay
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class Controller {
    @GetMapping("/v1/benchmark")
    suspend fun getHelloSleep(@RequestParam delay: Long): String {
        delay(delay)
        return "";
    }
}